apps/authentication-service/ARCHITECTURE.md

```diff
@@ -0,0 +1,33 @@
+# Authentication Architecture
+
+Below is a diagram of the authentication architecture for communication between a customer-hosted UCW-App instance, and
+the UCP-hosted Services.
+
+![Authentication Architecture](./assets/ucw-auth-architecture.png)
+
+The following is an outline of the process used to allow a customers’ self-hosted ucw-app to communicate with the
+UCP-hosted services (ucp-hosted-apps), such as Institutions, Performance, Preferences, etc.
+
+This process holds to the goal of not requiring UCP services to always be online in order for the customers’ ucw-app to
+function properly.
+
+However, in order for our customers to update their caches with the latest data from UCP services, they must create an
+account and get their required client id/secret values.
+
+If they choose not to update their caches, or if for some reason any of these UCP services are down, then the ucw-app
+will continue to function as expected due to the code shipping with default caches (institution, preferences, etc.).
+
+The process is as follows:
+
+1. A UCP customer will register via the UCP-UI Dashboard (URL TBD).
+1. The user will request access to generate API Keys.
+1. As part of the API Key creation process, the UCP Authentication service will create a new Application on the Auth0
+   platform. The Application ID is stored in the Auth0 User’s `app_metadata` object, so that we know which application a
+   user is associated with.
+1. This new Application will have an associated `client_id`, and `client_secret`, which the customer will be able to get
+   from the Auth UI Dashboard. They will put the id/secret values in the .env file of their ucw-app server instance.
+1. These values will reside in the `.env` vars `UCP_CLIENT_ID` and `UCP_CLIENT_SECRET`, respectively. The customer is
```

Comment:
I'd prefer not to name them so we don't have a potential syncing problem. We could point to the environment readme file